python/hail/typecheck/check.py

```diff
@@ -374,6 +374,12 @@ def lazy():
 
 char = CharChecker()
 
+table_key_type = transformed(
```

Comment:
this might be a little nicer as:
```
nullable(
  oneof(
    sequenceof(str),
    transformed((str, lambda x: [x]))))
```
```

Reply:
Cool, thanks. I still haven't completely groked how these typecheckers compose.